.circleci/config.yml

```diff
@@ -25,3 +25,124 @@ jobs:
           command: pipenv run mkdocs build --clean --strict --verbose
       - store_artifacts:
           path: site
+
+# Auto changelog collector
+  github-changelog-generator:
+    working_directory: ~/build
+    docker:
+      - image: ferrarimarco/github-changelog-generator:1.14.3
+    steps:
+      - setup_remote_docker:
+           version: 17.11.0-ce
+      - checkout
+      - run:
+           name: Build changelog
+           working_directory: ~/build
+           command: |
+             if git log -1 --pretty=%s | grep Merge* ; then
+             if git log -1 --pretty=%b | grep release ; then
+             echo "New release, do nothing"
```

Comment:
Is this check necessary? I wonder if there could be a situation where PR title would include the word "release" (with a different meaning).

Reply:
This check is necessary to make sure that the changelog will only generate on merged PR's. It will skip for commits and release PRs. For a release PR title we could have a controlled vocabulary or set of words to denote only a new specification release ("New main specification release"? with the new version followed i.e. New main specification release 1.2.3). I agree there can be a situation where a PR title could include the word "release" in it and mean something else. 

The release tag could work. The git describe --tags outputs the `"latest release tag"-# of commits since release-g"(commit string)"` . This can be parsed down to get the latest release tag, but I noticed that the git tag --list will output whatever tag you provide it, so the test will be true for all cases. I may not see how the test will skip the changelog generation for a new release?

Reply:
Try in some project with tags (say 0.1.0):

```Bash
for REF in 0.1.0 master; do
    git checkout -q $REF
    DESC=$(git describe --tags)
    if [ "$(git tag --list $DESC)" = "$DESC" ]; then
        echo "We're on a tag: $DESC"
    else
        echo "Not a tag, after all: $DESC"
    fi
done
```

In Pybids, for example:

```
We're on tag: 0.1.0
Not a tag, after all: 0.6.5-229-gbd23eaa
```

This is because `git tag --list 0.6.5-229-gbd23eaa` doesn't match any tag, and produces an empty string.

Reply:
Oh, I see. You plan to do release prep PRs. In which case, I would recommend using a template like "REL: <version>" for the PR name (or similar), and then looking for the "REL: " string in the merge commit. Otherwise you'll have to be vigilant about any PRs that have "release" somewhere in the title, which may show up innocently. "REL: " is much less likely to happen by accident.